Lib/test/datetimetester.py

```diff
@@ -886,6 +886,26 @@ def as_integer_ratio(self):
             with self.assertRaises(ValueError):
                 timedelta() * get_bad_float(bad_ratio)
 
+    def test_issue31752(self):
+        # The interpreter shouldn't crash because divmod() returns negative
+        # remainder.
+        class BadInt(int):
+            def __mul__(self, other):
+                return Prod()
+
+        class Prod:
```

Comment:
Maybe inherit from int as well?

Reply:
No, it shouldn't be inherited from int, otherwise `__radd__` would not be called.

Reply:
Oh ok.